### DISCUSSION TITLE
Can't make semantic-release work on protected branch

### DISCUSSION THREAD

**Discussion by felixjb** [semantic-release/semantic-release#2450] [Release workflow]
Hello!

I am trying to use semantic-release in my Continuous Delivery workflow in my repository, [Testify](https://github.com/felixjb/testify), but the action always fails because the remote does not allow the action to push a commit into a protected branch (main, in this case).

Now I don't know how to proceed. I would like to make my workflow as simple as clicking "Merge pull request" and then everything is done _automagically_. However, because GitHub does not even offer a feature to make the action an exception to this protection, I think maybe I will have to go another direction. 

How can I use the semantic-release, then?

**Comment by felixjb** [comment]
@travi @cycjimmy I would love your input on this.

**Comment by gethari** [comment]
Facing same problem in azure devops

**Comment by samiramkr** [comment]
Having the same problem!

**Comment by snlx** [comment]
Hi I've been trying to find an answer for this but couldn't find an answer anywhere and finally found this forum. did anyone manage to solve this issue ? I get this error  [remote rejected] HEAD -> master (pre-receive hook declined). 
Kindly help.

**Comment by travi** [comment]
if you have branch protection rules configured to prevent pushes to your release branch _and are making commits that are to be pushed back to that branch as part of the release_, semantic-release cannot perform any magic to bypass those rules on its own. 

if you decide to make commits as part of your release, despite [our recommendation](https://semantic-release.gitbook.io/semantic-release/support/faq#why-is-the-package.jsons-version-not-updated-in-my-repository), you have to account for that in your rule definition.

> because GitHub does not even offer a feature to make the action an exception to this protection

the situation around this is better than it was previously. our recommendation is to configure repository-rules instead of tranditional branch protection because it offers better flexibility for defining bypass rules more granularly than tranditional branch protection rules enabled
